src/elements/content-sidebar/activity-feed/comment/components/__tests__/BaseCommentMenu.test.js

```diff
@@ -0,0 +1,100 @@
+import { IntlProvider } from 'react-intl';
+import { fireEvent, render, type RenderResult } from '@testing-library/react';
+
+import { BaseCommentMenu } from '../BaseCommentMenu';
+import localize from '../../../../../../../test/support/i18n';
+
+import messages from '../../messages';
+
+jest.mock('react-intl', () => ({
+    ...jest.requireActual('react-intl'),
+    FormattedMessage: ({ defaultMessage }: { defaultMessage: string }) => <span>{defaultMessage}</span>,
+}));
+
+export const baseCommmentMenuDefaultProps: BaseCommentMenuProps = {
+    canDelete: true,
+    canEdit: true,
+    canResolve: true,
+    handleDeleteCancel: jest.fn(),
+    handleDeleteClick: jest.fn(),
+    handleDeleteConfirm: jest.fn(),
+    handleEditClick: jest.fn(),
+    handleMenuClose: jest.fn(),
+    handleStatusUpdate: jest.fn(),
+    isConfirmingDelete: false,
+    isResolved: false,
+    onSelect: jest.fn(),
+};
+
+const openMenu = (wrapper: RenderResult) => {
+    fireEvent.click(wrapper.queryByTestId('comment-actions-menu'));
+};
+
+// eslint-disable-next-line import/prefer-default-export
+export const getWrapper = props =>
+    render(
+        <IntlProvider locale="en">
+            <BaseCommentMenu {...baseCommmentMenuDefaultProps} {...props} />
+        </IntlProvider>,
+    );
+
+describe('elements/content-sidebar/ActivityFeed/comment/BaseCommentMenu', () => {
+    test.each`
+        prop            | value    | message
+        ${'canResolve'} | ${true}  | ${localize(messages.commentResolveMenuItem.id)}
+        ${'canResolve'} | ${false} | ${localize(messages.commentResolveMenuItem.id)}
+        ${'canEdit'}    | ${true}  | ${localize(messages.commentEditMenuItem.id)}
+        ${'canEdit'}    | ${false} | ${localize(messages.commentEditMenuItem.id)}
+        ${'canDelete'}  | ${true}  | ${localize(messages.commentDeleteMenuItem.id)}
+        ${'canDelete'}  | ${false} | ${localize(messages.commentDeleteMenuItem.id)}
+    `(`should render appropriate menu items when $prop is $value`, ({ prop, value, message }) => {
+        const wrapper = getWrapper({ [prop]: value });
+        openMenu(wrapper);
+        if (value) {
+            expect(wrapper.getByText(message)).toBeInTheDocument();
+        } else {
+            expect(wrapper.queryByText(message)).not.toBeInTheDocument();
+        }
+    });
+
+    test.each`
+        canResolve | isResolved | should
+        ${true}    | ${true}    | ${'should'}
+        ${false}   | ${true}    | ${'should NOT'}
+        ${true}    | ${false}   | ${'should NOT'}
+        ${false}   | ${false}   | ${'should NOT'}
+    `(
+        `$should render unresolve menu item when canResolve us $canResolve and isResolved is $isResolved`,
+        ({ canResolve, isResolved, should }) => {
+            const wrapper = getWrapper({ canResolve, isResolved });
+            const message = localize(messages.commentUnresolveMenuItem.id);
+            openMenu(wrapper);
+            if (should === 'should') {
```

Comment:
We typically want to avoid introducing logic in our tests. Best to make these two sets of independent tests.

Reply:
I think I disagree. As I wrote this today, I recalled our conversation about this a couple years back. I agreed at the time that it was better to separate them; but when I thought of this `should` param this morning, I thought it was a pretty nice solution. It's not exactly fearsome logic :)

Reply:
I was thinking about this. I definitely see the point of excluding, as much as possible, logic from tests since that new logic is then untested. Something as simple as this though, I'm on the fence about. I see the value in not wanting to set a precedent for untested logic in unit tests. But on the other hand, I like that these test are slightly easier to read and maintain since they aren't duplicated-ish. Both tradeoffs seem extremely minor, so kind of hard to decide. Could happily go your way :)

Reply:
That's not a bad idea @patlm

Reply:
@patim, that works well for cases where we're making assertions about the state of something that is rendered but not so well when we're asserting whether or not something is rendered at all, I think.